src/query/functions/src/scalars/map.rs

```diff
@@ -229,6 +241,123 @@ pub fn register(registry: &mut FunctionRegistry) {
         |map, _| map.len() as u64,
     );
 
+    registry.register_function_factory("map_delete", |_, args_type| {
+        if args_type.len() < 2 {
+            return None;
+        }
+
+        let (map_key_type, _map_value_type) = match args_type[0].remove_nullable() {
+            DataType::Map(box DataType::Tuple(type_tuple)) => {
+                if type_tuple.len() == 2 {
+                    (type_tuple[0].clone(), type_tuple[1].clone())
+                } else {
+                    return None;
+                }
+            }
+            DataType::EmptyMap => (DataType::EmptyMap, DataType::EmptyMap),
+            _ => {
+                return None;
+            }
+        };
+
+        let mut in_key_types: Option<DataType> = None;
+
+        for arg_type in args_type.iter().skip(1) {
+            match arg_type {
+                DataType::Array(box key_type) => {
+                    if map_key_type == DataType::EmptyMap {
+                        if let Some(in_key_type) = &in_key_types {
+                            if in_key_type != key_type {
+                                return None;
+                            }
+                        } else {
+                            in_key_types = Some(key_type.clone());
+                        }
+                    } else if *key_type != map_key_type {
+                        return None;
+                    }
+                }
+                DataType::Null => {
+                    return None;
+                }
+                _ => {
+                    return None;
+                }
+            }
+        }
+
+        let return_type = args_type[0].clone();
+
+        Some(Arc::new(Function {
+            signature: FunctionSignature {
+                name: "map_delete".to_string(),
+                args_type: args_type.to_vec(),
+                return_type: return_type.clone(),
+            },
+            eval: FunctionEval::Scalar {
+                calc_domain: Box::new(|_, args_domain| {
+                    FunctionDomain::Domain(args_domain[0].clone())
+                }),
+                eval: Box::new(move |args, _ctx| {
+                    let input_length = args.iter().find_map(|arg| match arg {
+                        ValueRef::Column(col) => Some(col.len()),
+                        _ => None,
+                    });
+
+                    let mut output_map_builder =
+                        ColumnBuilder::with_capacity(&return_type, input_length.unwrap_or(1));
+
+                    let delete_key_list = prepare_delete_key_list(args, input_length);
```

Comment:
this should be removed, we need to get `delete_key_list` for each row by the index.

Reply:
I wanted to clarify one point regarding the suggested solution. The `key` is currently considered as `HashSet<ScalarRef::Array(ArrayColumn<T>)>`, but for processing the map_delete operation, we need `HashSet<T>`. Would it be acceptable to keep it as `HashSet<T>`, or should I modify the logic to use `HashSet<ScalarRef::Array(ArrayColumn<T>)>` instead?